modules/apps/callbacks/types/callbacks.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v7/modules/core/05-port/types"
+	ibcexported "github.com/cosmos/ibc-go/v7/modules/core/exported"
+)
+
+// PacketUnmarshalerIBCModule is an interface that combines the IBCModule and PacketDataUnmarshaler
+// interfaces to assert that the underlying application supports both.
+type PacketUnmarshalerIBCModule interface {
+	porttypes.IBCModule
+	porttypes.PacketDataUnmarshaler
+}
+
+// CallbackData is the callback data parsed from the packet.
+type CallbackData struct {
```

Comment:
What do you think of having enough information here that we can just pass it directly to contractKeeper without any additional information.

If we add a callback type enum, we could even compress to a single interface function